components/eamxx/src/physics/mam/eamxx_mam_aci_process_interface.cpp

```diff
@@ -147,65 +206,62 @@ void MAMAci::run_impl(const double dt) {
+  Kokkos::parallel_for(team_policy, KOKKOS_LAMBDA(const haero::ThreadTeam &team) {
+    const int icol = team.league_rank();
+    // FIXME Is this the correct boundary condition for tke at the surface?
+    tke_(icol,nlev_) = 0;
+    Kokkos::parallel_for(Kokkos::TeamThreadRange(team, 0u, nlev_), KOKKOS_LAMBDA(int kk) { 
+      tke_(icol,kk) = (3.0/2.0)*w_sec_(icol,kk);
+    });
+  });
    //---------------------------------------------------------
    // Compute subgrid scale velocities(wsub, wsig and wsubice)
    //---------------------------------------------------------
   
-    // More refined computation of sub-grid vertical velocity
-    // Set to be zero at the surface by initialization.
-    /*
-    NOTE: All the inputs are available to compute wsub, wsig and wsubice
-          "min_max_bound" is present in MAM4xx  utils
-
-    Fortran code:
-    wsub(:ncol,:top_lev-1)  = wsubmin
-    wsubice(:ncol,:top_lev-1) = 0.001_r8
-    wsig(:ncol,:top_lev-1)  = 0.001_r8
-
-    do kk = top_lev, pver
-       do icol = 1, ncol
-          wsub(icol,kk)  = sqrt(0.5_r8*(tke(icol,kk) + tke(icol,kk+1))*(2._r8/3._r8))
-          wsig(icol,kk)  = min_max_bound(0.001_r8, 10._r8, wsub(icol,kk))
-          wsubice(icol,kk) = min_max_bound(0.2_r8, 10._r8, wsub(icol,kk))
-          wsub(icol,kk)  = max(wsubmin, wsub(icol,kk))
-       end do
-    end do
-    */
-
+  // More refined computation of sub-grid vertical velocity
+  // Set to be zero at the surface by initialization.
+  Kokkos::parallel_for(team_policy, KOKKOS_LAMBDA(const haero::ThreadTeam &team) {
+    const int icol = team.league_rank();
+    Kokkos::parallel_for(Kokkos::TeamThreadRange(team, 0u, top_lev_), KOKKOS_LAMBDA(int kk) { 
+      wsub_(icol,kk)  = wsubmin;
+      wsubice_(icol,kk) = 0.001;
+      wsig_(icol,kk)  = 0.001;
+    });
+    Kokkos::parallel_for(Kokkos::TeamThreadRange(team, top_lev_, nlev_), KOKKOS_LAMBDA(int kk) { 
+      wsub_(icol,kk)  = haero::sqrt(0.5*(tke_(icol,kk) + tke_(icol,kk+1))*(2.0/3.0));
+      wsig_(icol,kk)  = mam4::utils::min_max_bound(0.001, 10.0, wsub_(icol,kk));
+      wsubice_(icol,kk) = mam4::utils::min_max_bound(0.2, 10.0, wsub_(icol,kk));
+      wsub_(icol,kk)  = haero::max(wsubmin, wsub_(icol,kk));
+    });
+  });
 
     //---------------------------------------------------------
     // Compute subgrid mean updraft velocity (w2)
     //---------------------------------------------------------
   
-    /*
-    NOTE: All inputs are available. "subgrid_mean_updraft" is not ported yet but it is a very small routine
-    Fortran code:
-    w2(1:ncol,1:pver) = 0._r8
-    call subgrid_mean_updraft(ncol, w0, wsig, &!in
-         w2) !out
-    */
+    Kokkos::parallel_for(team_policy, KOKKOS_LAMBDA(const haero::ThreadTeam &team) {
+      const int icol = team.league_rank();
+      Kokkos::parallel_for(Kokkos::TeamThreadRange(team, 0u, nlev_), KOKKOS_LAMBDA(int kk) { 
+        w2_(icol,kk) = subgrid_mean_updraft(w0_(icol,kk), wsig_(icol,kk));
+      });
+  });
```

Comment:
@singhbalwinder :  Something of note:  There is a call to the Fortran function nucleate_ice_cam_calc which looks simple but there are quite a few arrays packed into the state_q mega-array parameter.  This function became the mam4xx::NucleateIce aerosol process which requires an Atmosphere, Prognostics, Diagnostics and Tendencies classes to execute and will need quite a few more of the add_field<Required> calls to set up and execute.